typing/typeclass.ml

```diff
@@ -1191,6 +1197,17 @@ and class_expr cl_num val_env met_env scl =
           cl_env = val_env;
           cl_attributes = scl.pcl_attributes;
          }
+  | Pcl_open (ovf, lid, e) ->
+      let used_slot = ref false in
+      let (path, new_val_env) = !Typecore.type_open ~used_slot ovf val_env scl.pcl_loc lid in
+      let (_path, new_met_env) = !Typecore.type_open ~used_slot ovf met_env scl.pcl_loc lid in
+      let cl = class_expr cl_num new_val_env new_met_env e in
+      rc {cl_desc = Tcl_open (ovf, path, lid, new_val_env, cl);
+          cl_loc = scl.pcl_loc;
+          cl_type = cl.cl_type;
+          cl_env = val_env;
```

Comment:
My understanding was that the environment here is just there so that we can properly interpret identifiers.
So any of the two environments is ok (the new one has just more information).
At least when it was introduced, the goal was not to be able to use this information to compile source code.
If we want to do that, we should define a more precise semantics.